README.md

```diff
@@ -202,6 +202,37 @@ When one model inherits from another, ElasticRecord makes some assumptions about
 
 These can all be overridden.  For instance, it might be desirable for the child documents to be in a separate index.
 
+### Join fields
+ElasticSearch provides support for join queries by providing a mechanism for declaring a join field that specifies the parent-child relationship between different types of records on the same index.
+ElasticRecord provides a short-(but not-so-short)-cut for declaring the mapping:
+
+```ruby
+class State
+  include ElasticRecord::Model
+end
+
+class Country
+  include ElasticRecord::Model
+
+  has_es_children(
+    join_field: 'pick_a_name_for_the_join_field',
+    children: ::ElasticRecord::Model::Joining::JoinChild.new(klass: State)
+  )
+end
+```
+
+`has_es_children` accepts an optional `name` argument, with a sane default. In the above example, it would default to `country`. The name can later be used to construct `has_parent` queries.
+The `join_field` will be used as the name of a method that gets defined both on the parent and on all children (and grandchildren), in addition to being the name of the mapping for the join field.  Be sure this doesn't conflict with the name of an existing method.  You are unlikely to need to use this name directly.
+
+`::ElasticRecord::Model::Joining::JoinChild.new`, too, accepts optional arguments with sane defaults:
+* `name`: In the above example, it would default to `state`. The name can later be used to construct `has_child` queries.
+* `children`: Can be another instance of `::ElasticRecord::Model::Joining::JoinChild` or an Array of them. Defaults to an empty Array.  Theoretically, an arbitrary number of layers of parent-child joins can be achieved this way.
+* `parent_id_accessor`: A method to call on the child to retrieve the ID of the parent. Can be a proc, which will be executed in the context of the child object, or the name of a method to be called on the child object.  In the above example, it would default to `country_id`.
+
+Note: Creating, deleting and updating mapping on the index must be handled via the Top-Level parent.  Running `rake index:create CLASS=State` has no effect.
+
+Note: The `load_from_source` configuration is not currently supported for indexes with a join field, though theoretically it could be implemented.
```

Comment:
```suggestion
`::ElasticRecord::Model::Joining::JoinChild.new` optional arguments:
* `name`: defaults to the snake case version of the value provided to `klass` (e.g. `state` in the example above). Can be used to construct `has_child` queries.
* `children`: Another instance of `::ElasticRecord::Model::Joining::JoinChild` or an Array of instances. Defaults to an empty Array.  Theoretically, an arbitrary number of layers of parent-child joins can be achieved this way.
* `parent_id_accessor`: Determines how the ID of the parent is retrieved. Can be a proc, which will be executed in the context of the child object, or a symbol corresponding to the name of a method defined on the child object.  In the above example, it would default to `country_id`.

Notes:
* Creating, deleting and updating mapping on the index must be handled via the Top-Level parent.  Running `rake index:create CLASS=State` has no effect.
* The `load_from_source` configuration is not yet supported for indices with a join field.
```

Reply:
I'm ok with this except for changing `currently` to `yet`.